File: leetcode/tree/count_complete_tree_nodes.py
from common import TreeNode
from typing import List

class Solution:
    def __init__(self):
        self.count = 0
    def countNodes(self, root: TreeNode) -> int:
        def helper(root):
            if root:
                self.count += 1
            else:
                return
            helper(root.left)
            helper(root.right)
        helper(root)
        return self.count

if __name__ == "__main__":
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.left = TreeNode(4)
    root.left.right = TreeNode(5)
    root.right.left = TreeNode(6)
    root.right.right = TreeNode(7)
    print(Solution().countNodes(root))
    

File: leetcode/tree/count_complete_tree_nodes.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func countNodes(root *TreeNode) int {
	if root == nil {
		return 0
	}
	var res int
	helper(root, &res)
	return res
}

func helper(root *TreeNode, res *int) {
	if root == nil {
		return
	} else {
		*res++
	}
	helper(root.Left, res)
	helper(root.Right, res)
}

func main() {
	root := &TreeNode{3, nil, nil}
	root.Left = &TreeNode{9, nil, nil}
	root.Right = &TreeNode{20, nil, nil}
	root.Right.Left = &TreeNode{15, nil, nil}
	root.Right.Right = &TreeNode{7, nil, nil}
	out := countNodes(root)
	fmt.Println(out)
}
